### PR TITLE
gh-105509: Simplify the implementation of `typing.Annotated`

### LINKED ISSUE #105509: Simplify the implementation of `typing.Annotated`
# Feature or enhancement

`typing.Annotated` is currently implemented as a class, but doesn't need to be. All other objects like it in the `typing` module are implemented as instances of `typing._SpecialForm`, and we can do the same here. This simplifies the code a lot (making it easier to maintain in the future), and should also be _slightly_ more performant.

### Linked PRs
* gh-105510

### PR DESCRIPTION
**Pull request by AlexWaygood**
Closes #105509

* Issue: gh-105509

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2023-06-08-15-56-45.gh-issue-105509.YIG57j.rst [hunk 1]
```diff
@@ -0,0 +1,3 @@
+``typing.Annotated`` is now implemented as an instance of
```
Comment (JelleZijlstra) [inline Misc/NEWS.d/next/Library/2023-06-08-15-56-45.gh-issue-105509.YIG57j.rst]:
```suggestion
:data:`typing.Annotated` is now implemented as an instance of
```

**Review by sobolevn — COMMENTED** [review]
Quick note: right now it is possible to use `Annotated` in `isinstance(1, typing.Annotated)`. However, it does not make much sense anyway:

```
>>> isinstance(typing.Annotated[int, 1], typing.Annotated)
False
```

With this patch applied it will raise a `TypeError`.

I don't think that it is a big deal.

**Comment by AlexWaygood** [comment]
Yeah. You can think up ways that this could break compatibility (and as such, I definitely don't think it should be backported). But I don't think we're changing anything about the public API for this symbol here, so nothing about the contract is being broken.

**Review by hauntsaninja — APPROVED** [review]
Looks good to me! Couldn't find any isinstance(..., Annotated) in a quick search

**Pull request merged**